src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -46,5 +51,46 @@ def contract_update_list_success(updated_resource, resource_client):
     # should be able to use the created model
     # to read since physical resource id is immutable
     _create_request, _created_model, _update_request, updated_model = updated_resource
+    test_primary_identifier_not_updated(
```

Comment:
ideally we could just make this its own test instead of checking for it twice. the fixture is the same for both contract_update_list_success and contract_update_read_success so you might be better off making a new test

Reply:
I think it makes sense to remove the **get_primary_identifier** method. This can be replaced by a public variable **primary_identifier_paths**